src/libxtp/gwbse/bse_operator.cc

```diff
@@ -46,21 +46,36 @@ Eigen::RowVectorXd BSE_OPERATOR<cqp, cx, cd, cd2>::row(int index) const {
 
 template <int cqp, int cx, int cd, int cd2>
 Eigen::RowVectorXd BSE_OPERATOR<cqp, cx, cd, cd2>::Hx_row(int index) const {
+  if (_Hx_cache[index].size() > 0) {
+    return std::move(_Hx_cache[index]);
+  }
   int auxsize = _Mmn.auxsize();
   vc2index vc = vc2index(0, 0, _bse_ctotal);
-  Eigen::VectorXd Hrow = Eigen::RowVectorXd::Zero(_bse_size);
+
   const int vmin = _opt.vmin - _opt.rpamin;
   const int cmin = _bse_cmin - _opt.rpamin;
   int v1 = vc.v(index);
   int c1 = vc.c(index);
-  const Eigen::VectorXd Mmn1Tc1 = _Mmn[v1 + vmin].row(c1 + cmin);
+  int cache_size = 10;
```

Comment:
could we make the cache size an option somehow ? Let say we have huge matrices and we don't want to cache, would be nice to set the size to 0.

Reply:
Yeah that is a good point. So I briefly benchmarked a bit and after 10 the effect seemed quite small. The memory need for 10 eigenvectors is actually also not that large. So I would argue, that the option confuses more than is actually helpful. I understand your point, but I would leave it as it is, if the need at some point arises, adding it is no problem